blueprint/core/blueprint_CanvasView.h

```diff
@@ -123,8 +123,11 @@ namespace blueprint
                         //TODO: Implement fillStyle pattern
                         //TODO: Implement fillStyle gradient
 
-                        const auto colourHex = args.arguments[0].toString();
-                        properties.fillStyle.setColour(juce::Colour::fromString(colourHex));
+                        const auto colorString = args.arguments[0].toString();
+                        juce::Colour colour = juce::Colour::fromString(colorString);
+                        colour = juce::Colours::findColourForName(colorString, colour);
```

Comment:
I'm :+1: for named color lookup, but I think we should do it on the javascript side.

My general philosophy is that we should write absolutely minimal native C++ when we can, because if we can ship helpful change without asking end users to recompile, I think that's a big win (especially for something like SOUL, where it would require that SOUL update and recompile their binaries _and_ for all SOUL end users to go get that update).

So, via that philosphy, I think our color approach should generally be: drastically simplify the language that our native API speaks (maybe we only ever take color strings of the form `ffaabbcc`, and then it's javascript business for getting from various color types `hsl`, `rgba`, `#hex` etc to the unified native color type).

Maybe it's worth me explaining that philosophy in the contributors guide? What do you think?

Reply:
awesome thanks for feedback @nick-thompson

Ahh yup that's smart then easy as `npm update react-juce` for updates. Yeah sure quick note in contributors guide couldn't hurt.

So for the named colors should we maintain our own map of JUCE names to hex value in JS land then? If so any particular place you'd like to keep them? Seems like we may need to create our own Colors class in JS to manage the conversion logic and mappings etc.

Reply:
Yeah totally agreed. Advising this as the preferred approach in the contributors guide sounds great. I too tried to do too much in C++ in my first contributions.

Reply:
> Ahh yup that's smart then easy as npm update react-juce for updates.

Exactly :)

> So for the named colors should we maintain our own map of JUCE names to hex value in JS land then? If so any particular place you'd like to keep them? Seems like we may need to create our own Colors class in JS to manage the conversion logic and mappings etc.

Yea, should be easy enough. Actually I just took a quick peek at npm to see if anything existed already, and check this out: https://www.npmjs.com/package/color-string. I wasn't aware of this before, but I think we could simplify much of our javascript side of things with this guy as well. That should solve taking named colors as well as hsl/a, rgb/a, etc and then we can map the resulting numeric array back into our own JUCE-hex style:

```js
var numbers = colorString.get.rgb('blue'); // [0, 0, 255, 1]
var hex = numbers.map(numericToHex);
var juceHex = `${hex[3]}${hex[0]}${hex[1]}${hex[2]}`;
```

Reply:
> Yea, should be easy enough. Actually I just took a quick peek at npm to see if anything existed already, and check this out: https://www.npmjs.com/package/color-string. I wasn't aware of this before, but I think we could simplify much of our javascript side of things with this guy as well. That should solve taking named colors as well as hsl/a, rgb/a, etc and then we can map the resulting numeric array back into our own JUCE-hex style:
> 
> ```js
> var numbers = colorString.get.rgb('blue'); // [0, 0, 255, 1]
> var hex = numbers.map(numericToHex);
> var juceHex = `${hex[3]}${hex[0]}${hex[1]}${hex[2]}`;
> ```
Nice find! Shame I didn't find it early haha. Yup looks like this will do everything we need and named colors seems to match JUCE 1 for 1 with the [color-name](https://github.com/colorjs/color-name/blob/master/index.js) package. 
I'll look into adding that tomorrow.